src/mmw/js/src/modeling/models.js

```diff
@@ -237,6 +240,120 @@ var ProjectModel = Backbone.Model.extend({
+             * possible scenarios: the old piece might be a BMP, might
+             * be a reclassification, or might represent an area where
+             * BMPs and reclassification overlap. */
+            else {
+                var newOldIntersection;
+
+                try {
+                    newOldIntersection = turfIntersect(oldPiece.shape, newPiece.shape);
+                } catch(e) {
+                    /* Once again, this "can only happen" if
+                     * oldPiece.shape is undefined, but make it
+                     * explicit. */
+                    oldPiece.shape = undefined;
+                    newOldIntersection = undefined;
+                }
+
+                /* New overlap pieces are born here. */
+                if ((newOldIntersection !== undefined) && (turfArea(newOldIntersection) > 0.33)) {
+                    var oldPieceShape = oldPiece.shape, // save a copy, need this for later
+                        overlapPiece = {
+                            name: both,
+                            shape: newOldIntersection,
+                            value: {}
+                        };
+
+                    /* compute overlap piece and add to array */
+                    if ((oldPiece.name === both) && (newPiece.name === reclass)) {
+                        overlapPiece.value.bmp = oldPiece.value.bmp;
+                        overlapPiece.value.reclass = newPiece.value;
+                    } else if ((oldPiece.name === both) && (newPiece.name === bmp)) {
+                        overlapPiece.value.bmp = newPiece.value;
+                        overlapPiece.value.reclass = oldPiece.value.reclass;
+                    } else if (oldPiece.name === reclass) {
+                        overlapPiece.value.bmp = newPiece.value;
+                        overlapPiece.value.reclass = oldPiece.value;
+                    } else {
+                        overlapPiece.value.bmp = oldPiece.value;
+                        overlapPiece.value.reclass = newPiece.value;
+                    }
+                    pieces.push(overlapPiece);
+
+                    /* remove the overlapping portion from both parents */
+                    try {
+                        oldPiece.shape = turfErase(oldPiece.shape, newPiece.shape);
+                    } catch(e) {
+                        /* Once again, should only happen if
+                         * oldPiece.shape is already undefined. */
+                        oldPiece.shape = undefined;
+                    }
+                    try {
+                        newPiece.shape = turfErase(newPiece.shape, oldPieceShape);
+                    } catch(e) {
+                        /* This can (should) never happen. */
+                    }
+                }
+            }
+        }
+
+        /* add remenants of new piece */
+        pieces.push(newPiece);
+
+        /* remove pieces that did not survive */
+        pieces = _.filter(pieces, function(p) {
```

Comment:
This anonymous function is causing lint errors and the build to fail:

```
> jshint $(find js -name *.js)

js/src/modeling/models.js: line 348, col 10, Don't make functions within a loop.

1 error
```

Consider moving this out of the loop and defining at the beginning with all other local variables.